src/main/java/org/sonarsource/sonarlint/ls/backend/BackendServiceFacade.java

```diff
@@ -44,10 +46,15 @@ public BackendInitParams getInitParams() {
     return initParams;
   }
 
-  public void initOnce() {
+  public void initOnce(Map<String, ServerConnectionSettings> connections) {
     if (initialized) return;
+    var sqConnections = BackendService.extractSonarQubeConnections(connections);
+    var scConnections = BackendService.extractSonarCloudConnections(connections);
+    initParams.setSonarQubeConnections(sqConnections);
+    initParams.setSonarCloudConnections(scConnections);
     backend.initialize(toInitParams(initParams));
     initialized = true;
+    backend.didChangeConnections(connections);
```

Comment:
This can't work because you returned early from the method (`if (initialized) return;`)

Reply:
Also I would rename the method because it's not a `initOnce` only

Reply:
Fixed.